WorkloadManagementSystem/Executor/JobScheduling.py

```diff
@@ -86,12 +86,23 @@ def optimizeJob( self, jid, jobState ):
     # If the user has selected any site, filter them and hold the job if not able to run
     if userSites:
       if jobType not in self.ex_getOption( 'ExcludedOnHoldJobTypes', [] ):
-        sites = self._applySiteFilter( userSites, banned = wmsBannedSites )
-        if not sites:
-          if len( userSites ) > 1:
-            return self.__holdJob( jobState, "Requested sites %s are inactive" % ",".join( userSites ) )
-          else:
-            return self.__holdJob( jobState, "Requested site %s is inactive" % userSites[0] )
+        result = self.__jobDB.getUserSitesTuple( userSites )
+        if not result[ 'OK' ]:
+          return S_ERROR( "Problem checking userSites for tuple of active/banned/invalid sites" )
+
+        userSites, banned_sites, invalid_sites = result['Value']
```

Comment:
To be purist, DIRAC naming convention is to use Camel names for all the variables, i.e. bannedSites and invalidSites